cuqi/distribution/_truncated_normal.py

```diff
@@ -47,7 +48,47 @@ def __init__(self, mean=None, std=None, low=-np.Inf, high=np.Inf, is_symmetric=F
         self.high = high
 
         # Init underlying normal distribution
-        self._normal = Normal(self.mean, self.std, is_symmetric=is_symmetric, **kwargs)
+        self._normal = Normal(self.mean, self.std, is_symmetric=True, **kwargs)
+
+    @property
+    def mean(self):
+        """ Mean of the distribution """
+        return self._mean
+
+    @mean.setter
+    def mean(self, value):
+        self._mean = force_ndarray(value, flatten=True)
+        if hasattr(self, '_normal'):
+            self._normal.mean = self._mean
+
+    @property
+    def std(self):
+        """ Std of the distribution """
+        return self._std
+
+    @std.setter
+    def std(self, value):
+        self._std = force_ndarray(value, flatten=True)
+        if hasattr(self, '_normal'):
+            self._normal.std = self._std
+
```

Comment:
I wonder if we can simplify the logic a bit so we do not need to store local `_mean` and `_std`. I mean if we can set the initializer, and mean and std setters and getters to be something like this:

```python
    def __init__(self, mean=None, std=None, low=-np.Inf, high=np.Inf, is_symmetric=False, **kwargs):
        # Init from abstract distribution class
        super().__init__(is_symmetric=is_symmetric, **kwargs) 
        # Init specific to this distribution
        self.low = low
        self.high = high
        
        self._normal = Normal(mean, std, is_symmetric=True, **kwargs)

    @property
    def mean(self):
        """ Mean of the distribution """
        return self._normal.mean

    @mean.setter
    def mean(self, value):
        self._normal.mean = force_ndarray(value, flatten=True)

    @property
    def std(self):
        """ Std of the distribution """
        return self._normal.std

    @std.setter
    def std(self, value):
        self._normal.std = force_ndarray(value, flatten=True)

```

Reply:
This simplified code seems to cause errors in initializing the Gibbs sampler, so after talking with @amal-ghamdi, we decided to keep it as is.

Reply:
Maybe we can then just keep it as it is, unless @nabriis has other suggestions

Reply:
No I think we cannot do it much simpler